src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerEnclaveProvider.java

```diff
@@ -96,3 +253,59 @@ long getCounter() {
         return counter.getAndIncrement();
     }
 }
+
+
+final class EnclaveSessionCache {
+    Hashtable<String, EnclaveCacheEntry> sessionCache;
+
+    EnclaveSessionCache() {
+        sessionCache = new Hashtable<>(0);
+    }
+
+    void addEntry(String servername, String attestationUrl, BaseAttestationRequest b, EnclaveSession e) {
+        System.out.println("Adding session " + e.getSessionID().toString() + " with key: " + servername + attestationUrl);
+        sessionCache.put(servername+attestationUrl, new EnclaveCacheEntry(b, e,attestationUrl));
+    }
+
+    EnclaveCacheEntry getSession(String key) {
+        EnclaveCacheEntry e = sessionCache.get(key);
+        if (null != e && e.expired()) {
+            sessionCache.remove(key);
+            return null;
+        }
+        return e;
+    }
+}
+
+
+class EnclaveCacheEntry {
+    private static final long EIGHT_HOUR_IN_MILLIS = 28800000;
+
+    private BaseAttestationRequest bar;
+    private EnclaveSession es;
+    private String attestationUrl;
+    private long timeCreatedInMillis;
+
+    EnclaveCacheEntry(BaseAttestationRequest b, EnclaveSession e, String url) {
+        bar = b;
+        es = e;
+        attestationUrl = url;
+        timeCreatedInMillis = Instant.now().getEpochSecond();
+    }
+
+    public boolean expired() {

```

Comment:
it doesn't really matter since they're all implicitly public, but seems inconsistent to specify that just for this one method and not the others